### PR TITLE
depositETH with referral

### PR DESCRIPTION
**Pull request by danoctavian**
This solves all gameability issues since it pairs up msg.sender and referrer on-chain.

Also solves any data storage issue and passing any referral data between off-chain systems.

Adding referrer to existing Deposit event changes its signature - this makes it slightly more work to parse out the referral events since you need to index both and look transaction ids.

### CODE REVIEW

File: src/ReferralDepositAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { IynETH } from "src/interfaces/IynETH.sol";
```
[the remark this answers is not in the document]
Comment (danoctavian) [reply src/ReferralDepositAdapter.sol:4]: addressed!

**Comment by kahuna099** [comment]
lgtm  :raised_hands:

**Review by kahuna099 — APPROVED** [review]

**Review by gearcat0 — APPROVED** [review]

File: script/Actors.sol [hunk 2]
```diff
@@ -105,7 +107,8 @@ contract ActorAddresses {
                 LSD_RESTAKING_MANAGER: mainnetWallets.YNDev,
                 STAKING_NODE_CREATOR: mainnetWallets.YNDev,
                 POOLED_DEPOSITS_OWNER: 0xE1fAc59031520FD1eb901da990Da12Af295e6731,
-                PAUSE_ADMIN: mainnetWallets.YNDev
+                PAUSE_ADMIN: mainnetWallets.YNDev,
+                REFERAL_PUBLISHER: holeskyWallets.YNDev
```
Comment (marchev) [inline script/Actors.sol:111]: This should probably be `mainnetWallets.YNDev`

**Comment by xhad** [comment]
@danoctavian we should be ready to deploy this on mainnet and setup legacy referrals sync via the yieldnest-points. I would need the contract deployed to start mainnet referral indexing. Happy to take on this deployment, or perhaps we collab sometime early this week

**Comment by danoctavian** [comment]
> @danoctavian we should be ready to deploy this on mainnet and setup legacy referrals sync via the yieldnest-points. I would need the contract deployed to start mainnet referral indexing. Happy to take on this deployment, or perhaps we collab sometime early this week

Addressed

https://github.com/yieldnest/yieldnest-protocol/blob/feature/referral-code-deposit/deployments/ReferralDepositAdapter-1.json

ReferralDepositAdapter has been deployed for mainnet here.

Please verify the deployment is correct.

ProxyAdmin owner should be YNSecuritCouncil and the REFERRAL_PUBLISHER_ROLE should be controlled by YNDev.

**Pull request merged**